src/cosmosdb-preview/azext_cosmosdb_preview/custom.py

```diff
@@ -72,6 +77,212 @@ def _handle_exists_exception(cloud_error):
+    
+    mongocluster_restore_parameters = MongoClusterRestoreParameters(
+        point_in_time_utc=restore_point_in_time_utc,
+        source_resource_id=restore_source_resource_id,
+    )
+
+    node_group_spec = NodeGroupSpec(
+        sku= shard_node_sku,
+        disk_size_gb= shard_node_disk_size_gb,
+        enable_ha=shard_node_ha,
+        name= shard_node_name,
+        kind=shard_kind,
+        node_count= shard_node_count
+    )
+    
+    node_group_specs = [node_group_spec]
+    mongodb_cluster = MongoCluster(
+        location=location,
+        tags=tags,
+        create_mode=create_mode,
+        restore_parameters= mongocluster_restore_parameters,
+        administrator_login=administrator_login,
+        administrator_login_password=administrator_login_password,
+        server_version=server_version,
+        node_group_specs=node_group_specs)
+
+    return client.begin_create_or_update(resource_group_name, cluster_name, mongodb_cluster)
+
+def cli_cosmosdb_mongocluster_update(client,
+                                    resource_group_name,
+                                    cluster_name,
+                                    administrator_login=None,
+                                    administrator_login_password=None,
+                                    tags=None,
+                                    create_mode=CreateMode.DEFAULT.value,
+                                    server_version="5.0",
+                                    shard_node_sku=None,
+                                    shard_node_ha=None,
+                                    shard_node_disk_size_gb=None, 
+                                    shard_node_name=None,
+                                    shard_kind=NodeKind.SHARD.value):
+
+    '''Updates an Azure Cosmos DB Mongo Cluster '''
+
+    mongo_cluster_resource = client.get(resource_group_name, cluster_name)
+
+    # user name and password should be updated together
+    if administrator_login is None and administrator_login_password is not None:
+          raise InvalidArgumentValueError('Both administrator_login and administrator_login_password should be updated together.')
+    
+    if administrator_login is not None and administrator_login_password is None:
+          raise InvalidArgumentValueError('Both administrator_login and administrator_login_password should be updated together.')
+  
+    if administrator_login_password is None:
+        administrator_login_password = mongo_cluster_resource.administrator_login_password
+
+    # Resource location is immutable
+    location = mongo_cluster_resource.location
+
+    if server_version is None:
+        server_version = mongo_cluster_resource.server_version
+    if tags is None:
+        tags = mongo_cluster_resource.tags
+    if create_mode is None:
+        create_mode= mongo_cluster_resource.create_mode
```

Comment:
nit-picking: tidy up the spacing around the equals here and in the other points in the code so that it is consistent

Reply:
fixed